internal/services/controller/handlers/handler.go

```diff
@@ -53,7 +53,7 @@ func (h *handler) OnDelete(obj interface{}) {
 }
 
 func (h *handler) handle(e castai.EventType, obj interface{}) {
-	if h.filters != nil && !h.filters.Apply(e, obj) {
+	if !h.filters.Apply(e, obj) {
```

Comment:
Maybe just change this check to `len(h.filters) > 0 && !h.filters.Apply(e, obj)` so we wouldn't have pass `filters.Filters{{}}` everywhere where filters are not needed?